pyspecProcScripts/calc_error.py

```diff
@@ -0,0 +1,104 @@
+from .simple_functions import select_pathway
+import numpy as np
+
+
+# {{{ Functions to calculate var and mean of nan-masked data
+#     Pulled from time_domain_noise.py example
+def _masked_mean_multi(x, axis=None):
+    "Calculate the mean of nan-masked data on a 1D axis"
+    assert axis is not None
+
+    def masked_mean(x):
+        return np.mean(x[np.isfinite(x)])
+
+    return np.apply_along_axis(masked_mean, axis, x)
+
+
+def _masked_var_multi(x, axis=None):
+    """Calculates the variance along a 1D axis.
+    Note, it calculates the variance of the real of the data"""
+
+    def masked_var(x):
+        if np.iscomplex(axis) and sum(abs(np.imag(axis))) > 1e-7:
+            raise ValueError(
+                "Data claims to be complex but has a negligible imaginary part"
+            )
+        else:
+            return np.var(x[np.isfinite(x)], ddof=1)
+
+    if axis is not None:
+        return np.apply_along_axis(masked_var, axis, x)
+    else:
+        return masked_var(x)
+
+
+# }}}
+def calc_masked_error(
+    s,
+    excluded_frqs=None,
+    excluded_pathways=None,
+    direct="t2",
+    indirect="nScans",
```

Comment:
it's weird that this defaults to `"nScans"`, but the integration function defaults to `"vd"`.  Pick one.

Reply:
checked -s tick with vd